src/Kovan-DssSpell.sol

```diff
@@ -52,154 +32,110 @@ contract DssSpellAction is DssAction {
-            "RWA002_A_INPUT_CONDUIT", RWA002_A_INPUT_CONDUIT
-        );
-        DssExecLib.setChangelogAddress(
-            "RWA002_A_OUTPUT_CONDUIT", RWA002_A_OUTPUT_CONDUIT
-        );
-        DssExecLib.setChangelogVersion("1.2.11");
-
-        // --------------------- Remove ESM_BUG ---------------------
-        address MCD_ESM          = DssExecLib.getChangelogAddress("MCD_ESM_ATTACK");
-        address MCD_ESM_BUG      = DssExecLib.getChangelogAddress("MCD_ESM_BUG");
-        address MCD_END          = DssExecLib.end();
-
-        DssExecLib.deauthorize(MCD_END, MCD_ESM_BUG);
-
-        CHANGELOG.removeAddress("MCD_ESM_BUG");
-        CHANGELOG.removeAddress("MCD_ESM_ATTACK");
+        address MCD_VAT          = DssExecLib.vat();
+        address MCD_CAT          = DssExecLib.cat();
+        address MCD_DOG          = DssExecLib.getChangelogAddress("MCD_DOG");
+        address MCD_VOW          = DssExecLib.vow();
+        address MCD_POT          = DssExecLib.pot();
+        address MCD_SPOT         = DssExecLib.spotter();
+        address MCD_END_OLD      = DssExecLib.end();
+        address MCD_ESM_OLD      = DssExecLib.getChangelogAddress("MCD_ESM");
+        address MCD_CLIP_LINK_A  = DssExecLib.getChangelogAddress("MCD_CLIP_LINK_A");
+        address PIP_LINK         = DssExecLib.getChangelogAddress("PIP_LINK");
+
+        // ------------------  END  ------------------
+
+        // Set contracts in END
+        DssExecLib.setContract(MCD_END,  "vat", MCD_VAT);
+        DssExecLib.setContract(MCD_END,  "cat", MCD_CAT);
+        DssExecLib.setContract(MCD_END,  "dog", MCD_DOG);
+        DssExecLib.setContract(MCD_END,  "vow", MCD_VOW);
+        DssExecLib.setContract(MCD_END,  "pot", MCD_POT);
+        DssExecLib.setContract(MCD_END, "spot", MCD_SPOT);
+
+        // Authorize the new END in contracts
+        DssExecLib.authorize(MCD_VAT, MCD_END);
+        DssExecLib.authorize(MCD_CAT, MCD_END);
+        DssExecLib.authorize(MCD_DOG, MCD_END);
+        DssExecLib.authorize(MCD_VOW, MCD_END);
+        DssExecLib.authorize(MCD_POT, MCD_END);
+        DssExecLib.authorize(MCD_SPOT, MCD_END);
+
+        // Set wait time in END
+        Fileable(MCD_END).file("wait", EndAbstract(MCD_END_OLD).wait());
+
+        // Deauthorize the old END in contracts
+        DssExecLib.deauthorize(MCD_VAT, MCD_END_OLD);
+        DssExecLib.deauthorize(MCD_CAT, MCD_END_OLD);
+        DssExecLib.deauthorize(MCD_DOG, MCD_END_OLD);
+        DssExecLib.deauthorize(MCD_VOW, MCD_END_OLD);
+        DssExecLib.deauthorize(MCD_POT, MCD_END_OLD);
+        DssExecLib.deauthorize(MCD_SPOT, MCD_END_OLD);
+
+        // Deauthorize the old END from all the FLIPS
+        // Authorize the new END in all the FLIPS
+        bytes32[] memory ilks = IlkRegistryAbstract(ILK_REGISTRY).list();
+        for (uint256 i = 0; i < ilks.length; i++) {
+            bytes32 ilk = ilks[i];
+            if (IlkRegistryAbstract(ILK_REGISTRY).class(ilk) < 3) {
```

Comment:
For the current rwa collaterals, the `xlip` is set to `address(0)`

We could probably save a whole bunch of lookups by just requesting `addr = reg.xlip(ilk)` and then doing an `if (addr != address(0)) { <deauth>; <auth>; }`

Not a big deal on kovan but might be worthwhile to use the pattern now in case we use it as a mainnet pattern.

Reply:
Yeah it's kovan. I haven't been saving gas at all. That's why I just repeated the loop as well.